content/courses/native-onchain-development/hello-world-program.md

```diff
@@ -13,156 +13,152 @@ description:
+specific "data type" known at compile time. In cases where multiple types are
+possible, we must add a type annotation to our variables.
 
 In the example below, we create a function named `process_instruction` that
 requires the following arguments:
 
-- `program_id` - required to be type `&Pubkey`
-- `accounts` - required to be type `&[AccountInfo]`
-- `instruction_data` - required to be type `&[u8]`
+- `program_id` - required to be of type `&Pubkey`.
+- `accounts` - required to be of type `&[AccountInfo]`.
+- `instruction_data` - required to be of type `&[u8]`.
 
 Note the `&` in front of the type for each argument listed in the
-`process_instruction` function. In Rust, `&` represents a ”reference” to another
-variable. This allows you to refer to some value without taking ownership of it.
-The “reference” is guaranteed to point to a valid value of a particular type.
-The action of creating a reference in Rust is called “borrowing”.
+`process_instruction` function. In Rust, `&` represents a "reference" to another
+variable, allowing you to refer to some value without taking ownership of it.
+The reference is guaranteed to point to a valid value of a particular type. The
+action of creating a reference in Rust is called “borrowing.”
 
 In this example, when the `process_instruction` function is called, a user must
 pass in values for the required arguments. The `process_instruction` function
-then references the values passed in by the user, and guarantees that each value
-is the correct data type specified in the `process_instruction` function.
+then references the values passed in by the user, guaranteeing that each value
+is the correct data type specified in the function.
 
 Additionally, note the brackets `[]` around `&[AccountInfo]` and `&[u8]`. This
-means that the `accounts` and `instruction_data` arguments expect “slices” of
-types `AccountInfo` and `u8`, respectively. A “slice” is similar to an array
-(collection of objects of the same type), except the length is not known at
+indicates that the `accounts` and `instruction_data` arguments expect "slices"
+of types `AccountInfo` and `u8`, respectively. A “slice” is similar to an array
+(a collection of objects of the same type), except the length is not known at
 compile time. In other words, the `accounts` and `instruction_data` arguments
 expect inputs of unknown length.
 
 ```rust
-fn process_instruction(
+pub fn process_instruction(
     program_id: &Pubkey,
     accounts: &[AccountInfo],
     instruction_data: &[u8],
 )
 ```
 
-We can then have our functions return values by declaring the return type using
-an arrow `->` after the function.
+We can also have our functions return values by declaring the return type using
+an arrow -> after the function.
 
 In the example below, the `process_instruction` function will now return a value
-of type `ProgramResult`. We will go over this in the next section.
+of type `ProgramResult`. We’ll go over this in the next section.
```

Comment:
```suggestion
of type `ProgramResult`. We'll go over this in the next section.
```

Reply:
I'm not totally sure there's any benefit to smart quotes. The rest of the code uses regular quotes so I think we should stick to that, or change it everywhere. Feel free to change my mind!

Reply:
This was from old content only, I didn't notice. I have changed it to regular quotes across the lesson.